### PR TITLE
 Add BALLERINA_MAX_POOL_SIZE

### PR DESCRIPTION
**Pull request by vinok88**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by vinok88** [comment]
> I think we should use "BALLERINA_MAX_POOL_SIZE".

Done.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/scheduling/Scheduler.java [hunk 1]
```diff
@@ -67,9 +68,27 @@
     }
 
     private AtomicInteger totalStrands = new AtomicInteger();
-    private final int numThreads;
+    /**
+     * By default number of threads = (Available logical Processors * 2).
```
Comment (sameerajayasoma) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/scheduling/Scheduler.java]: Shall we change this as well?

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/BLangConstants.java [hunk 2]
```diff
@@ -72,6 +72,7 @@
     public static final String USER_REPO_SRC_DIRNAME = "src";
     public static final String USER_REPO_OBJ_DIRNAME = "obj";
     public static final String USER_REPO_METADATA_DIRNAME = "metadata";
+    public static final String BAL_MAX_POOL_SIZE_ENV = "BALLERINA_MAX_POOL_SIZE";
```
Comment (sameerajayasoma) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/BLangConstants.java]: Shall we change this variable name to be consistent with other variables in this file?

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BRunUtil.java [hunk 3]
```diff
@@ -432,7 +432,7 @@ protected static void invokePackageInit(CompileResult compileResult, String pack
                 }
             };
 
-            Scheduler scheduler = new Scheduler(4, false);
+            Scheduler scheduler = new Scheduler( false);
```
Comment (Kishanthan) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BRunUtil.java]: Formatting issue here "new Scheduler( false);" this has an extra space character

**Pull request merged**